marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/flow/job-query-lib.sjs

```diff
@@ -59,6 +59,43 @@ function findStepResponses(query) {
   return response;
 }
 
+function valuesExist(values) {
+  return !(values || values.length);
+}
+
+function findJobDocuments({start, pageLength, jobId, status, startTimeBegin, startTimeEnd, endTimeBegin, endTimeEnd, user}) {
+  const queries = [cts.collectionQuery('Job')];
+  if (valuesExist(jobId)) {
+    queries.push(cts.jsonPropertyValueQuery('jobId', jobId));
+  }
+  if (valuesExist(status)) {
+    queries.push(cts.jsonPropertyValueQuery('jobStatus', status));
+  }
+  if (valuesExist(startTimeBegin)) {
+    queries.push(cts.rangeQuery(cts.elementReference(xs.QName("startTime"), ["type=dateTime"]), '>=', startTimeBegin));
+  }
+  if (valuesExist(startTimeEnd)) {
+    queries.push(cts.rangeQuery(cts.elementReference(xs.QName("startTime"), ["type=dateTime"]), '<=', startTimeEnd));
+  }
+  if (valuesExist(endTimeBegin)``) {
+    queries.push(cts.rangeQuery(cts.elementReference(xs.QName("endTime"), ["type=dateTime"]), '>=', endTimeBegin));
+  }
+  if (valuesExist(endTimeEnd)) {
+    queries.push(cts.rangeQuery(cts.elementReference(xs.QName("endTime"), ["type=dateTime"]), '<=', endTimeEnd));
+  }
+  if (valuesExist(user)) {
+    queries.push(cts.jsonPropertyValueQuery('user', user));
+  }
+  const finalQuery = cts.andQuery(queries);
+  const total = cts.estimate(finalQuery);
+  return {
+    total,
+    start,
+    pageLength,
+    results: fn.subsequence(cts.search(finalQuery), start, pageLength)
```

Comment:
Thoughts on using jsearch instead? In theory that should be a simpler interface to code against.